fdbclient/NativeAPI.actor.cpp

```diff
@@ -3100,6 +3111,51 @@ ACTOR static Future<Void> commitDummyTransaction( Database cx, KeyRange range, T
 	}
 }
 
+ACTOR static Future<Optional<CommitID>> getCommitVersion(Database cx, TransactionOptions options, Version version,
+                                                         UID txID) {
+	state Transaction tr(cx);
+	state Key key = keyForTransaction(version, txID);
+	state int retries = 0;
+	state Future<Optional<Value>> minVersion;
+	state Future<Optional<Value>> versionStamp;
+	state Error err;
+	loop {
+		try {
+			tr.options = options;
+			tr.setOption(FDBTransactionOptions::TRANSACTION_IS_IDEMPOTENT);
+			versionStamp = tr.get(key);
+			minVersion = tr.get(transactionIDMinVersion);
+			wait(success(versionStamp) && success(minVersion));
+			if (minVersion.get().present()) {
+				auto v = decodeMinTxVersionValue(minVersion.get().get());
+				if (v > version) {
+					err = commit_unknown_result();
+					break;
+				}
+			}
+			if (versionStamp.get().present()) {
+				if (versionStamp.get().get().size() < 10) {
+					err = transaction_aborted();
+					break;
+				}
+				CommitID res;
+				std::tie(res.version, res.txnBatchId) = decodeTxVersion(versionStamp.get().get());
+				res.metadataVersion = cx->metadataVersionCache[cx->mvCacheInsertLocation].second;
+				return res;
+			} else {
+				tr.set(key, LiteralStringRef(""));
```

Comment:
If this commit goes through, is it just going to trigger the `transaction_aborted` error? If that's the intent, the description of the error should probably change (currently "aborted due to system error").

Reply:
I confused myself a bit here... Thanks for pointing this out. This should not set an empty string but either set nothing and just throw `commit_unknown_result` or it should set `0` as a version stamp.